common/unsafe/src/test/java/org/apache/spark/unsafe/PlatformUtilSuite.java

```diff
@@ -157,4 +157,11 @@ public void heapMemoryReuse() {
     Assertions.assertEquals(1024 * 1024 + 7, onheap4.size());
     Assertions.assertEquals(obj3, onheap4.getBaseObject());
   }
+
+  @Test
+  public void cleanerCreateMethodIsDefined() {
+    // Regression test for SPARK-45508: we don't expect the "no cleaner" fallback
+    // path to be hit in normal usage.
+    Assertions.assertTrue(Platform.cleanerCreateMethodIsDefined());
```

Comment:
If we want to backport this change then we will have to prepare a separate PR because https://github.com/apache/spark/pull/43074 changed the JUnit assertions helper names so the same source can't compile in all branches.

Reply:
> If we want to backport this change then we will have to prepare a separate PR because #43074 changed the JUnit assertions helper names so the same source can't compile in all branches.

sorry for making this change ...